Preserve Data in .json file and simplify logic

Yazino12 opened the pull request:
### In this pull:

- Preserve data for MusicAlbum and Genre
- Simplify code to a simpler logic

**This closes #29**

Yazino12 committed:
Preserve Data in .json file and simplify logic

aimalamiri commented on classes/app.rb:
```diff
@@ -5,10 +5,11 @@
-    @manage_labels = ManageLabels.new(@items)
-    @manage_genres = ManageGenres.new(@items)
+    @manage_labels = ManageLabels.new(@labels)
+    @manage_genres = ManageGenres.new(@labels)
```
We need a variable for gener as well. `@genres` but you're passing the `@labels`

Yazino12 replied:
@aimalamiri, Thanks for pointing this issue out. I will implement this right away.

aimalamiri commented on classes/managers/manage_genres.rb:
```diff
@@ -1,46 +1,34 @@
 class ManageGenres
-  def initialize(items)
-    @items = items
+  def initialize(labels)
+    @labels = labels
+    @file_writer = FileWriter.new('labels.json')
+    @file_reader = FileReader.new('labels.json')
```
Wouldn't be better to rename the `labels` with `genres` because it's all related to genres?

aimalamiri commented on classes/managers/manage_genres.rb:
```diff
@@ -1,46 +1,34 @@
-      end
-    else
-      puts 'Add item first!'
+    load_labels
+    @labels.each do |label|
+      puts "Genre - #{label.name}"
```
Wouldn't be better to rename the `labels` with `genres` because it's all related to genres?

aimalamiri commented on classes/managers/manage_genres.rb:
```diff
@@ -1,46 +1,34 @@
+    puts "The label #{genre.name} has been added successfully!"
+  end
+
+  def load_labels
+    @labels = @file_reader.read_data
```
Wouldn't be better to rename the labels with genres because it's all related to genres?

Yazino12 replied:
@aimalamiri Addressed these changes ✔

aimalamiri reviewed (changes requested):
## Good job!

## Status: Changes Required

Good job @Yazino12 you're code is clean and readable. Although it can be improved somewhat by following the instructions in the comment below.

_Please check the comments._

Yazino12 committed:
Implement the required changes
Modify Genres storage array to be Genres
Access Genres data from Genres storage

aimalamiri reviewed (approved):
**Hello @Yazino12  👋🏾**

### PROJECT STATUS: APPROVED 🟢🟢🟢

**To recap:**

- No linter errors ✔
- Data preserved correctly ✔
- Good design ✔

![](https://media.giphy.com/media/8UF0EXzsc0Ckg/giphy.gif)

Congratulations on successfully completing this task 🎉🎉🍾🍾. There is nothing to say, but it's time to merge it 🚀.
